### PR TITLE
Add molecule role testing to etcd role

### PR DESCRIPTION
**Pull request by chrisplo**
Molecule
https://molecule.readthedocs.io/en/latest/

Provides repeatable role testing by defining scenarios of host
configuration and playbooks.

This first scenario sets up a three node etcd cluster.  Later
iterations can do deployment validation, lint checks, and idempotency
checks.

Signed-off-by: Chris Plock <chrisplo@cisco.com>

### CODE REVIEW

**Comment by retr0h** [comment]
Awesome 👍  v2 getting used here.

**Review by tiewei — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -11,3 +11,9 @@ test-cleanup:
 
 test-test:
 	CONTIV_ANSIBLE_TAGS="prebake-for-test" vagrant provision
+
+test-etcd:
+	cd roles/etcd && virtualenv venv && . venv/bin/activate \
+		&& pip install --upgrade pip \
+		&& pip install -r molecule-requirements.txt \
+		&& molecule converge; molecule destroy
```
Comment (vhosakot) [inline Makefile]: why not `&& molecule converge && molecule destroy` ?

Comment (chrisplo) [reply Makefile]: if this is CI, then it cleans up, though darn, i lose the exit status, lemme fix

File: roles/etcd/molecule/default/destroy.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      set_fact:
+        instance_conf: {}
+
+    - name: Dump instance config
+      copy:
+        # NOTE(retr0h): Workaround for Ansible 2.2.
+        #               https://github.com/ansible/ansible/issues/20885
+        content: "{{ instance_conf | to_json | from_json | molecule_to_yaml | molecule_header }}"
+        dest: "{{ molecule_instance_config }}"
+      when: server.changed | bool
```
Comment (vhosakot) [inline roles/etcd/molecule/default/destroy.yml]: extra char, can be removed in the next PR :)

**Review by vhosakot — APPROVED** [review]

**Pull request merged**